hawkmoth/__init__.py

```diff
@@ -230,11 +230,33 @@ class CAutoUnionDirective(_AutoCompoundDirective):
 class CAutoEnumDirective(_AutoCompoundDirective):
     _docstring_types = [docstring.EnumDocstring]
 
+def _deprecate(conf, old, new, default=None):
+    if conf[old]:
+        logger = logging.getLogger(__name__)
+        logger.warning(f'{old} is deprecated in favour of {new}')
+        conf[new] = conf[old]
+        del conf[old]
+        return conf[new]
+    return default
+
```

Comment:
Need to double check modification of the config does not break Sphinx caching the environment for subsequent runs.

Reply:
Hmm, I thought it gated the compilation on `conf.py` being changed, but it's a legitimate concern.

I didn't notice any difference with regards to skipping compilation steps or not when compiling our own docs. How do you usually check this?

Reply:
I think history has shown I usually don't. :p

Reply:
Played with it a bit, and I think it's okay.